stripe/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -379,22 +395,28 @@ internal class DefaultFlowController internal constructor(
         }
     }
 
-    private fun createPaymentSheetResult(
-        paymentIntentResult: PaymentIntentResult
+    private fun <T : StripeIntent> createPaymentSheetResult(
+        stripeIntentResult: StripeIntentResult<T>
     ): PaymentSheetResult {
-        val paymentIntent = paymentIntentResult.intent
+        val stripeIntent = stripeIntentResult.intent
         return when {
-            paymentIntent.status == StripeIntent.Status.Succeeded ||
-                paymentIntent.status == StripeIntent.Status.RequiresCapture -> {
+            stripeIntent.isConfirmed -> {
                 PaymentSheetResult.Completed
             }
-            paymentIntentResult.outcome == StripeIntentResult.Outcome.CANCELED -> {
+            stripeIntentResult.outcome == StripeIntentResult.Outcome.CANCELED -> {
                 PaymentSheetResult.Canceled
             }
-            paymentIntent.lastPaymentError != null -> {
+            stripeIntent is PaymentIntent && stripeIntent.lastPaymentError != null -> {
```

Comment:
In trying to elliminate this if-else on the intent type here are some thoughts:
1. Make a lastError function in Stripe intent
2. Just set the message to stripeIntent.lastErrorMessage().   None of the messages in PaymentSheetResult.Failed are intended to be printed to the user.